tests/integration/cli/commands/test_issue_2211.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os.path
+
+import pytest
+
+from pex.interpreter_constraints import InterpreterConstraint
+from pex.pep_440 import Version
+from pex.pip.version import PipVersion, PipVersionValue
+from pex.typing import TYPE_CHECKING
+from testing import run_pex_command
+from testing.cli import run_pex3
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+# N.B.: awscli==1.28.1 Only resolves with the pip-2020-resolver for Pip versions earlier than 22.
+# In the end this means only Pip >=20.3.2,<22 can run this test successfully. The underlying issue
+# comes from PyYAML and is documented here: https://github.com/yaml/pyyaml/issues/601
+@pytest.mark.parametrize(
+    "pip_version",
+    [
+        pytest.param(pip_version, id=str(pip_version))
+        for pip_version in PipVersion.values()
+        if Version("20.3.2") <= pip_version.version < Version("22")
```

Comment:
I'll leave this as-is but I just realized there are no `--pip-version` options in this range except for vendored Pip (20.3.4-patched); so only 1 test is generated here. In short, this fix only applies to that 1 version of Pip while using the pip-2020-resolver. All other versions of Pip just fail fast in a situation like this and you're out of options if the situation is not solvable with `--constraints` or you can't upgrade requirements past the problem.

Reply:
I think that's fine - the parametrization is informative, and if we ever added some other intermediate pip version for some reason this would not require tending.

Reply:
Agreed. As it turns out it was a bit wrong. the requirement is actually `>=20.3.2,!=20.3.3,<22` where 20.3.2 is a yanked version; so the requirement in practice really is just `>=20.3.4,<22`.